plugins/woocommerce/tests/php/src/Internal/AbilitiesApi/AbilitiesApiIntegrationTest.php

```diff
@@ -0,0 +1,267 @@
+<?php
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Tests\Internal\AbilitiesApi;
+
+/**
+ * Tests for the WordPress Abilities API integration with WooCommerce
+ *
+ * @since 10.4.0
+ */
+class AbilitiesApiIntegrationTest extends \WC_Unit_Test_Case {
+
+	/**
+	 * Set up the test environment.
+	 */
+	public function set_up() {
+		parent::set_up();
+
+		// Trigger the abilities API initialization if it hasn't been done yet.
+		if ( did_action( 'abilities_api_init' ) === 0 ) {
+			/**
+			 * This action is documented in vendor/wordpress/abilities-api/bootstrap.php
+			 *
+			 * @since 11.0.0
+			 */
+			do_action( 'abilities_api_init' );
+		}
+	}
+
+	/**
+	 * Test that core Abilities API classes are available.
+	 *
+	 * @group abilities-api
+	 */
+	public function test_core_classes_are_available() {
+		$this->assertTrue(
+			class_exists( 'WP_Ability' ),
+			'WP_Ability class should be available'
+		);
+		$this->assertTrue(
+			class_exists( 'WP_Abilities_Registry' ),
+			'WP_Abilities_Registry class should be available'
+		);
+	}
+
+	/**
+	 * Test that global Abilities API functions are available.
+	 *
+	 * @group abilities-api
+	 */
+	public function test_global_functions_are_available() {
+		$this->assertTrue(
+			function_exists( 'wp_register_ability' ),
+			'wp_register_ability() function should be available'
+		);
+		$this->assertTrue(
+			function_exists( 'wp_get_ability' ),
+			'wp_get_ability() function should be available'
+		);
+		$this->assertTrue(
+			function_exists( 'wp_get_abilities' ),
+			'wp_get_abilities() function should be available'
+		);
+		$this->assertTrue(
+			function_exists( 'wp_unregister_ability' ),
+			'wp_unregister_ability() function should be available'
+		);
+	}
+
+	/**
+	 * Test that we can register a simple ability.
+	 *
+	 * @group abilities-api
+	 */
+	public function test_can_register_ability() {
+		$ability_id = 'woocommerce-test/simple-test';
+
+		$result = wp_register_ability(
+			$ability_id,
+			array(
+				'label'            => 'Simple Test Ability',
+				'description'      => 'A simple test ability for unit testing',
+				'input_schema'     => array(
+					'type'       => 'object',
+					'properties' => array(
+						'message' => array(
+							'type' => 'string',
+						),
+					),
+				),
+				'output_schema'    => array(
+					'type'       => 'object',
+					'properties' => array(
+						'result' => array(
+							'type' => 'string',
+						),
+					),
+				),
+				'execute_callback' => function ( $input ) {
+					return array(
+						'result' => 'Test executed with: ' . ( $input['message'] ?? 'no message' ),
+					);
+				},
+			)
+		);
+
+		$this->assertInstanceOf( 'WP_Ability', $result, 'Ability registration should return a WP_Ability instance' );
+
+		// Clean up.
+		$this->cleanup_ability( $ability_id );
```

Comment:
Nitpick: in WP core, this type of cleanup is handled in the `tear_down` to account for edge cases where a failed assertion would terminate the execution, preventing cleanup. It's more of a practice that doesn't make a big difference in these types of tests, which should always pass.